### PR TITLE
use instanceof Function for the checking function

### PR DESCRIPTION
**Pull request by quarrant**

### CODE REVIEW

**Comment by quarrant** [comment]
@codeBelt review please

File: src/utils.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const isObjectWithProperties = (data: any): boolean => {
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export const isFunction = (functionToCheck: any): boolean => {
-  return functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';
+  return functionToCheck && functionToCheck instanceof Function;
```
Comment (codeBelt) [inline src/utils.ts:32]: Will this also handle GeneratorFunction?

https://stackoverflow.com/a/38510353

Comment (quarrant) [reply src/utils.ts:32]: Honestly, I didn't think about this... Also I don't remember any storage which may uses generators.

Comment (quarrant) [reply src/utils.ts:32]: So, I checked and it works same

Comment (codeBelt) [reply src/utils.ts:32]: cool, approved!

**Pull request merged**